forward/testing/backend_tracker.lua

```diff
@@ -0,0 +1,45 @@
+local ffi = require("ffi")
+local C = ffi.C
+local godefs = require("godefs")
+
+local BackendTracker = {}
+
+function BackendTracker:new()
+   return setmetatable({
+      backends = {},
+   }, {
+      __index = BackendTracker
+   })
+end
+
+function BackendTracker:add_backends(backends)
+   for _, b in ipairs(backends) do
+      godefs.AddBackend(b.name, b.addr, b.addr_len, b.health_check_type)
+      table.insert(self.backends, b)
+   end
+   C.usleep(1000)
+end
+
+function BackendTracker:clear_backends()
+   for _, b in ipairs(self.backends) do
+      godefs.RemoveBackend(b.name)
+   end
+   C.usleep(10000)
```

Comment:
Why are these sleeps necessary?

Reply:
The backend handling in Go is multithreaded, so it's necessary to wait for updates to propagate to Maglev whenever making changes to the backends before proceeding with the tests. I'm not sure why removing backends requires a longer sleep; possibly it takes some time to propagate channel closing in Go.